aspnetcore/grpc/interprocess.md

```diff
@@ -29,101 +35,18 @@ The preceding project file:
-
-Unix domain sockets connection factory example:
-
-```csharp
-public class UnixDomainSocketConnectionFactory
-{
-    private readonly EndPoint _endPoint;
-
-    public UnixDomainSocketConnectionFactory(EndPoint endPoint)
-    {
-        _endPoint = endPoint;
-    }
-
-    public async ValueTask<Stream> ConnectAsync(SocketsHttpConnectionContext _,
-        CancellationToken cancellationToken = default)
-    {
-        var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
-
-        try
-        {
-            await socket.ConnectAsync(_endPoint, cancellationToken).ConfigureAwait(false);
-            return new NetworkStream(socket, true);
-        }
-        catch
-        {
-            socket.Dispose();
-            throw;
-        }
-    }
-}
-```
-
-Using the custom connection factory to create a channel:
-
-```csharp
-public static readonly string SocketPath = Path.Combine(Path.GetTempPath(), "socket.tmp");
-
-public static GrpcChannel CreateChannel()
-{
-    var udsEndPoint = new UnixDomainSocketEndPoint(SocketPath);
-    var connectionFactory = new UnixDomainSocketConnectionFactory(udsEndPoint);
-    var socketsHttpHandler = new SocketsHttpHandler
-    {
-        ConnectCallback = connectionFactory.ConnectAsync
-    };
-
-    return GrpcChannel.ForAddress("http://localhost", new GrpcChannelOptions
-    {
-        HttpHandler = socketsHttpHandler
-    });
-}
-```
-
-Channels created using the preceding code send gRPC calls over Unix domain sockets. Support for other IPC technologies can be implemented using the extensibility in Kestrel and `SocketsHttpHandler`.
+## Inter-process communication transports
+
+gRPC calls between a client and server on different machines are usually sent over TCP sockets. TCP is a great choice for communicating across a network or the Internet. .NET supports multiple transports designed for IPC workloads.
+
+* [Unix domain sockets](https://wikipedia.org/wiki/Unix_domain_socket) is a widely supported IPC technology that's more efficient than TCP when the client and server are on the same machine. UDS is usable on Linux, macOS, and [Windows 10/Windows Server 2019 or later](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/).
+* [Named pipes](https://wikipedia.org/wiki/Named_pipe) are supported by all versions of Windows. Named pipes integrate well with [Windows security](/windows/win32/ipc/named-pipe-security-and-access-rights) to control client access to the pipe.
+
+Implement other IPC technologies using the extensibility in Kestrel and `SocketsHttpHandler`.
+
+## Configure client and server
+
+The client and server must be configured to use an IPC transport. For more information about configuring Kestrel and `SocketsHttpHandler` to use IPC:
```

Comment:
On first mention in a section, I try to state the acronyms because we don't know where the reader is coming from. They might land directly on this from an external link and not know what it stands for.

```suggestion
The client and server must be configured to use an inter-process communication (IPC) transport. For more information about configuring Kestrel and <xref:System.Net.Http.SocketsHttpHandler> to use IPC:
```